json-ld/src/main/scala/io/renku/jsonld/Flattener.scala

```diff
@@ -0,0 +1,112 @@
+/*
+ * Copyright 2020 Swiss Data Science Center (SDSC)
+ * A partnership between École Polytechnique Fédérale de Lausanne (EPFL) and
+ * Eidgenössische Technische Hochschule Zürich (ETHZ).
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.renku.jsonld
+
+import cats.syntax.all._
+import io.renku.jsonld.JsonLD.{JsonLDArray, JsonLDEntity, JsonLDEntityId, MalformedJsonLD}
+
+import scala.annotation.tailrec
+
+trait JsonLDArrayFlattener extends Flattener {
```

Comment:
Would that maybe work if that trait is named `JsonLDArrayFlatten`. I think it'd kind of match better the functionality of the trait that it introduces the `flatten` method.